nestjs/src/profile/dto/update-profile.dto.ts

```diff
@@ -142,50 +142,50 @@ class GeneralInfo {
   @ValidateNested()
   location: Location;
 
-  @ApiProperty({ required: false, nullable: true })
+  @ApiProperty({ required: false, nullable: true, type: [Education] })
   @IsOptional()
   @Type(() => Education)
   @IsArray()
   educationHistory?: Education[] | null;
 
-  @ApiProperty({ required: false, nullable: true })
+  @ApiProperty({ required: false, nullable: true, type: String })
   @IsOptional()
   @IsString()
   englishLevel?: EnglishLevel | null;
 }
 
 class Contacts {
-  @ApiProperty({ required: false, nullable: true })
+  @ApiProperty({ required: false, nullable: true, type: String })
   @IsOptional()
   @IsString()
   phone: string | null;
 
-  @ApiProperty({ required: false, nullable: true })
+  @ApiProperty({ required: false, nullable: true, type: String })
   @IsOptional()
   @IsString()
   email: string | null;
 
-  @ApiProperty({ required: false, nullable: true })
+  @ApiProperty({ required: false, nullable: true, type: String })
   @IsOptional()
   @IsString()
   epamEmail: string | null;
 
-  @ApiProperty({ required: false, nullable: true })
+  @ApiProperty({ required: false, nullable: true, type: String })
   @IsOptional()
   @IsString()
   skype: string | null;
 
-  @ApiProperty({ required: false, nullable: true })
+  @ApiProperty({ required: false, nullable: true, type: String })
   @IsOptional()
   @IsString()
   telegram: string | null;
 
-  @ApiProperty({ required: false, nullable: true })
+  @ApiProperty({ required: false, nullable: true, type: String })
   @IsOptional()
   @IsString()
   notes: string | null;
 
-  @ApiProperty({ required: false, nullable: true })
+  @ApiProperty({ required: false, nullable: true, type: String })
```

Comment:
just curious, won't it all be strings by definition?

Reply:
it fails to detect type if you use union `string | null`. It becomes `object` :(